core/filtermaps/filtermaps.go

```diff
@@ -610,7 +610,9 @@ func (f *FilterMaps) storeFilterMapRowsOfGroup(batch ethdb.Batch, mapIndices []u
 	if uint32(len(mapIndices)) != f.baseRowGroupLength { // skip base rows read if all rows are replaced
 		var ok bool
 		baseRows, ok = f.baseRowsCache.Get(baseMapRowIndex)
-		if !ok {
+		if ok {
```

Comment:
Should we also clone the slice in `getFilterMapRow`?

Reply:
We do clone `baseRow` and if required then we append `extRow` which is always retrieved from the database so it's guaranteed that the returned result does not share an underlying array with anything.